samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -345,6 +349,22 @@ def _clean_running_containers_and_related_resources(self) -> None:
         cast(WarmLambdaRuntime, self.lambda_runtime).clean_running_containers_and_related_resources()
         cast(RefreshableSamFunctionProvider, self._function_provider).stop_observer()
 
+    def _add_account_id_to_global(self):
+        """
+        Attempts to get the Account ID from the current session
+        If there is no current session, the standard parameter override for
+        AWS::AccountId is used
+        """
+        sts = boto3.client("sts")
+        try:
+            account_id = sts.get_caller_identity().get("Account")
+            if account_id:
+                if self._global_parameter_overrides is None:
+                    self._global_parameter_overrides = {}
+                self._global_parameter_overrides["AWS::AccountId"] = account_id
+        except TokenRetrievalError:
```

Comment:
I think the right exception type to catch is `botocore.exceptions.ClientError`, as explained [here](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#aws-service-exceptions)

Reply:
I tried using ClientError but that made all the tests fail. I think TokenRetrievalError is correct. Maybe ClientError is for failures instantiating/setting up the client.